### PR TITLE
CRIMAP-382 Tweaks to the schemas, and new definitions

### PR DESCRIPTION
**Pull request by zheileman**
## Description of change
MAAT team has expressed the library they are using can't handle "allOf, anyOf, oneOf".

So as the less disruptive solution, in the MAAT schema we are not going to use these, and instead flatten the "applicant" (which is the only place where these were used). There will be a bit of repetition but is not that bad as that allow us to keep evolving our schema separate from MAAT.

At the same time, I've extracted to shared definitions Interests Of Justice, and Provider details. These are unlikely to change so I think is safe to share these ones between our schema and MAAT schema. We can later change our minds.

Ensured some optional/mandatory attributes are marked as such.

## Link to relevant ticket
https://dsdmoj.atlassian.net/browse/CRIMAP-382

## Additional notes
There should be no breaking changes but I've bumped the patch version for explicitness.

### CODE REVIEW

File: schemas/1.0/general/provider.json [hunk 1]
```diff
@@ -0,0 +1,18 @@
+{
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "$id": "ministryofjustice/laa-criminal-legal-aid-schemas/main/schemas/1.0/general/provider.json",
+  "title": "Provider",
+  "description": "Holds common attributes of a provider",
+  "type": "object",
+  "properties": {
+    "office_code": { "type": "string" },
```
Comment (arthurashman) [inline schemas/1.0/general/provider.json:8]: Was it intentional to remove the minLength attr?

**Review by arthurashman — APPROVED** [review]
LGTM 👍

File: schemas/1.0/general/provider.json (hunk 1 above)
Comment (zheileman) [reply schemas/1.0/general/provider.json:8]: Intentional. It was introduced originally when we didn't have a proper LAA Portal, and one mock user returned 0 office codes leaving this as blank.
As we don't have that problem anymore, I'd rather not have this arbitrary validation (in theory office codes will be 6 chars length but not sure if we want to have that constraint at all in the schema).

**Review by willmcb — APPROVED** [review]
💥

**Pull request merged**